src/verifiers/documentStatus/didSignedDocumentStatus/didSignedDocumentStatus.ts

```diff
@@ -129,10 +55,14 @@ const verify: VerifierType["verify"] = async (_document, _option) => {
 
     // Check that all the issuers have signed on the document
     if (!document.proof) throw new Error("Document is not signed. Proofs are missing.");
-    const issuanceDeferred: IssuanceStatus[] = issuers.map((issuer: any) =>
+    const signatureVerificationDeferred: DidVerificationStatus[] = issuers.map((issuer: any) =>
       verifySignature({ merkleRoot, identityProof: issuer.identityProof, proof: document.proof, did: issuer.id })
     );
-    const issuance = await Promise.all(issuanceDeferred);
+    const issuance = await (await Promise.all(signatureVerificationDeferred)).map(({ verified, reason, did }) => ({
+      issued: verified,
+      did,
+      ...(reason && { reason }),
```

Comment:
https://github.com/Open-Attestation/oa-verify/pull/120/files#diff-43471af6de049beffa58e92b1126431bR43

The reason looks from this ? but this is not an iterator so it will fail

Reply:
This is to conditionally add key. Doing this to not break existing test that doesnt need to change. If I have just the reason key, it will be `reason: undefined`.

![image](https://user-images.githubusercontent.com/7406870/93177682-c067b100-f765-11ea-952f-02db6b43a3e2.png)